pkg/azureclients/armclient/util.go

```diff
@@ -123,7 +123,8 @@ func DoHackRegionalRetryDecorator(c *Client) autorest.SendDecorator {
 			// Hack: retry the regional ARM endpoint in case of ARM traffic split and arm resource group replication is too slow
 			// Empty content and 2xx http status code are returned in this case.
 			// Issue: https://github.com/kubernetes-sigs/cloud-provider-azure/issues/1296
-			emptyResp := (response.ContentLength == 0 || trimmed == "" || trimmed == "{}") && response.StatusCode >= 200 && response.StatusCode < 300
+			// Update: ContentLength == -1 is also considered because ARM can possibly return ContentLength -1 and StatusCode 200.
+			emptyResp := (response.ContentLength == 0 || response.ContentLength == -1 || trimmed == "" || trimmed == "{}") && response.StatusCode >= 200 && response.StatusCode < 300
```

Comment:
could you add some unit tests for the new retry logic?

Reply:
Sure